test/functional/lua/diagnostic_spec.lua

```diff
@@ -1749,6 +1749,18 @@ describe('vim.diagnostic', function()
           return _G.count_extmarks(_G.diagnostic_bufnr, _G.diagnostic_ns)
         end)
       )
+
+      eq(
+        true,
+        exec_lua(function()
+          vim.diagnostic.config({
+            float = function(_, _)
+              return { border = 'rounded' }
+            end,
+          })
+          return type(vim.diagnostic.config().float) == 'function'
```

Comment:
instead of asserting this (which doesn't really test much), can we assert `eq('rounded', ...)` ?

And before this assertion, check that border is NOT 'rounded', to make sure we are actually testing something.

Reply:
I have added in open_float